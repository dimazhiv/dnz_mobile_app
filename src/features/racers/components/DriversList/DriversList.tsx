import React from 'react';
import { FlatList, StyleSheet, View } from 'react-native';
import DriverItem from './DriverItem/DriverItemContainer';
import { DriversIds } from '../../../../types/NormalizedLoadedDrivers';

export type DriversListProps = {
  driversIds: DriversIds;
};

const renderItem = ({ item }: { item: string }) => <DriverItem driverId={item} />;

const renderSeparator = () => <View style={styles.separator} />;

function DriversList({ driversIds }: DriversListProps) {
  return (
    <FlatList
      style={styles.flatList}
      ItemSeparatorComponent={renderSeparator}
      data={driversIds}
      renderItem={renderItem}
      showsVerticalScrollIndicator={false}
    />
  );
}

const styles = StyleSheet.create({
  flatList: {
    paddingTop: 10,
    paddingBottom: 20
  },
  separator: {
    flex: 1,
    paddingTop: 20
  }
});

export default DriversList;
