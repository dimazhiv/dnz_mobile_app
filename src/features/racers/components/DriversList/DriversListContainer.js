import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { getDriversIds } from '../../selectors';
import DriversList from './DriversList';

function mapStateToProps(state) {
  return {
    driversIds: getDriversIds(state)
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({}, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(DriversList);
