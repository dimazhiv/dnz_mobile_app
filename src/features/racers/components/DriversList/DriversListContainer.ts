import { bindActionCreators, Dispatch } from 'redux';
import { connect } from 'react-redux';
import { getCurrentPageDriverIds } from '../../selectors';
import DriversList from './DriversList';
import { RootStateType } from '../../../../setupStore';

function mapStateToProps(state: RootStateType) {
  return {
    driversIds: getCurrentPageDriverIds(state)
  };
}

function mapDispatchToProps(dispatch: Dispatch) {
  return bindActionCreators({}, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(DriversList);
