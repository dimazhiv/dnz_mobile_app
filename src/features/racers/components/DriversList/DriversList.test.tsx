import React from 'react';
import { shallow } from 'enzyme';
import DriversList from './DriversList';

describe('Snapshot testing DriversList component', () => {
  test('correct component view', () => {
    const tree = shallow(<DriversList driversIds={['alonso', 'senna']} />);
    expect(tree).toMatchSnapshot();
  });
});
