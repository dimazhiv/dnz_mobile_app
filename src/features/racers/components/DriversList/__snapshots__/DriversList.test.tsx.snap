// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing DriversList component correct component view 1`] = `
<FlatList
  ItemSeparatorComponent={[Function]}
  data={
    [
      "alonso",
      "senna",
    ]
  }
  renderItem={[Function]}
  showsVerticalScrollIndicator={false}
  style={
    {
      "paddingBottom": 20,
      "paddingTop": 10,
    }
  }
/>
`;
